Plugins/SimpleFileBrowser/Scripts/FileBrowser.cs

```diff
@@ -2468,8 +2468,13 @@ private int FilenameInputToFileEntryIndex( string input, int startIndex, int len
 		{
 			for( int i = 0; i < validFileEntries.Count; i++ )
 			{
-				if( validFileEntries[i].Name.Length == length && input.IndexOf( validFileEntries[i].Name, startIndex, length ) == startIndex )
```

Comment:
Thank you for the PR! I'd rather use the overload of IndexOf that accepts StringComparison here and use platform dependent compilation directives to distinguish case insensitive platforms. E.g:

```csharp
#if UNITY_SOME_CASE_INSENSITIVE_PLATFORMS
if( ...modified if with StringComparison... )
#else
if( ...original if... )
#endif
```

There are no good answers to which StringComparison option to pick because while OrdinalIgnoreCase will fail for i-İ, CurrentCultureIgnoreCase will fail for i-I on Turkish culture (I don't think this is the only example of this issue). I think instead of dealing with StringComparisons, I may just keep my code as is. When you pass an initialFilename and you expect it to match an existing file, then you usually already know the cases of the letters in the filename so it's better to simply pass an initialFilename with the correct cases.

I'll keep the PR open at the moment and later on, I may change my mind and either pull it or close it.

Reply:
For sure makes sense! I do have an academic challenge on the hard coded directives (which I typically like). The OS does not determine the case sensitivity of the file system necessarily. Technically a Linux or MacOS system could mount a case insensitive NTFS, or a remote SMB. However I appreciate this is a corner case. 

I am very curious, as it's clear I do not work on a ton on international character sets - why do you favour the IndexOf comparison? It's just not something I have come across and am sure you have made a very deliberate choice to use it.

Thanks so much for reading through the Pull. I absolutely loved your plugin. It was exactly what I needed!

Reply:
In this instance, I'm using IndexOf to check if the substring of _input_ that starts at _startIndex_ and has a length of _length_ is equal to another string (_validFileEntries[i].Name_). Without IndexOf, the only alternatives I know of are: 1) using Substring which will cause GC allocations, 2) iterating over the characters in the specified range manually and comparing them with _validFileEntries[i].Name_'s characters which is unnecessary work.

I also use IndexOf with the _textComparer_. In that instance, the textComparer performs a case insensitive comparison that actually counts ı-i-I-İ letters the same which is very useful while filtering files by their names. Come to think of it, you could use textComparer in your case as well to perform case-insensitive comparison.

Reply:
@yasirkula Do you still interested in this kind of feature?
I'll probably go with conditional compilation directive for this feature. I don't want to directly modify the source code as much, but the method is non-virtual private so it can't be helped in my case

My use case is that I have another input to check whether a (save) file exists and that Windows file is case-insensitive, so I want to have an option to specify if this file browser will act the same way

Reply:
@aunpyz Save file overwrite dialog [is in my to-do list](https://github.com/yasirkula/UnitySimpleFileBrowser/issues/62). While implementing it, I'll probably need to check for case sensitivity, as well. To check the OS' case sensitivity, I'm planning to write a dummy FileBrowserCaseCheck.txt file to temporaryCachePath and then check if `File.Exists("filebrowsercasecheck.txt")` returns true.

Reply:
@yasirkula Thank you very much for shipping the feature this fast!
I'm not sure if I will upgrade to latest version as my project will end very soon and I don't want to risk anything right now, but it does have really nice feature there

Again, thank you!